enhancements/operator-framework-api/catalogue-check.md

```diff
@@ -0,0 +1,180 @@
+
+### Implementation Details/Notes/Constraints 
+
+Following bellow the check criteria which are proposed here for the new validator. Note that new rules and checks could be addressed in the future.
+
+#### OpenShift version supportability features
+
+The operator framework already has implemented some features that allow users to define the OCP version(s) that the operator bundle will support already. Also, another one which is the following annotation is planned to be implemented.
+
+**Annotation maxOpenShiftVersion**
+
+See [here](/enhancements/olm/max-openshift-versions-for-operators.md) a proposal for the operator bundles can use the following annotation:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    # Prevent upgrades to OpenShift Version 4.9
+    operators.coreos.com/maxOpenShiftVersion: "4.8"
+```
+
+**OCP Labels in the image index**
+
+By using the label `LABEL com.redhat.openshift.versions`, it is possible to define range criteria see:
+
+```yaml
+FROM scratch
+
+LABEL operators.operatorframework.io.bundle.mediatype.v1=registry+v1
+LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
+LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
+LABEL operators.operatorframework.io.bundle.package.v1=your-operator
+LABEL operators.operatorframework.io.bundle.channels.v1=your-channel
+LABEL operators.operatorframework.io.bundle.channel.default.v1=your-channel
+LABEL com.redhat.openshift.versions: v4.5-v4.7
+
+COPY manifests /manifests/
+COPY metadata /metadata/
+```
+
+The versioning scheme available in the above openshift versions label accepts:
+
+- Min version (e.g `v4.5`)
+- Range (e.g `v4.5-v4.7`)
+- A specific version (e.g `=v4.6`)
+
+Following some examples:
+
+| Label | Description |
+| ------ | ----- |
+| `LABEL com.redhat.openshift.versions:v4.5` | It means valid for OCP `4.5` and later versions |
+| `LABEL com.redhat.openshift.versions:=v4.5` | Valid **ONLY** for `4.5`
+| `LABEL com.redhat.openshift.versions:v4.5-v4.7` | Valid **ONLY** for `4.5`, `4.6` and `4.7` |
+
+**CSV spec for minimal k8s version**
+
+The spec `minKubeVersion` describe what is the minimal version of Kubernetes that is supported by the operator bundle, see:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+...
+spec:
+  ... 
+  minKubeVersion: 1.11.0
+```
+
+The above features can be used alone and/or combined.
+
+#### Criteria
+
+| Check | Type | Description |
+| ------ | ----- | ----- |
+|To ensure that the operator bundle specified the OCP versions which the project should be supported.|warning|Check if `maxOpenShiftVersion` in CSV **OR** the `com.redhat.openshift.versions` in the Dockerfile is used.
```

Comment:
see above, i don't consider these two things equivalent.

checking for both is reasonable, but the error/warning/implications of not having each are different.

and i suspect lots of operators are not going to specify maximum ocp versions, nor an upper bound on their ocp catalog version.  So we'll need to think about exactly what we want the error/warning/guidance to be here.

Reply:
I improved that and tried to clarify its better. Let me know if it makes more sense now.